ESP8266wifi.cpp

```diff
@@ -663,15 +667,33 @@ byte ESP8266wifi::readCommand(int timeout, const char* text1, const char* text2)
 }*/
 
 // Reads count chars to a buffer, or until delim char is found
-byte ESP8266wifi::readBuffer(char* buf, byte count, char delim) {
-    byte pos = 0;
-    while (_serialIn->available() && pos < count) {
-        if (_serialIn->peek() == delim)
-            break;
-        buf[pos++] = readChar();
-    }
-    buf[pos] = '\0';
-    return pos;
+//byte ESP8266wifi::readBuffer(char* buf, byte count, char delim) {
+    //byte pos = 0;
+    //while (_serialIn->available() && pos < count) {
+        //if (_serialIn->peek() == delim)
+            //break;
+        //buf[pos++] = readChar();
+    //}
+    //buf[pos] = '\0';
+    //return pos;
+//}
+
+// Read count chars to a buffer, or until delim char is found, or the request timed out.
+byte ESP8266wifi::readBuffer(char* buf, byte count, char delim, int timeout) {
+	byte pos = 0;
+    unsigned long stop = millis() + timeout;
+
+	while (pos < count) {
+		if(_serialIn->available()) {
+			if (_serialIn->peek() == delim)
+				break;
+			buf[pos++] = readChar();
+		}
+		if(timeout > 0 && millis() > stop)
+			break; //timed out.
+	}
+	buf[pos] = '\0';
+	return pos;
```

Comment:
Please just replace outdated code instead of commenting it out. Easier diff to read. And we trust in git for history. :smile:

Also. Something weird with indentation here. Use 4 spaces please.